test/e2e/common/node/pod_admission.go

```diff
@@ -65,6 +66,50 @@ var _ = SIGDescribe("PodOSRejection", framework.WithNodeConformance(), func() {
 	})
 })
 
+var _ = SIGDescribe("PodOSRejectionStatus", func() {
+	f := framework.NewDefaultFramework("pod-os-rejection-status")
+	f.NamespacePodSecurityLevel = admissionapi.LevelBaseline
+	ginkgo.Context("Kubelet", func() {
+		ginkgo.It("should reject pod when the node OS doesn't match pod's OS", func(ctx context.Context) {
+			linuxNode, err := findLinuxNode(ctx, f)
+			framework.ExpectNoError(err)
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wrong-pod-os",
+					Namespace: f.Namespace.Name,
+				},
+				Spec: v1.PodSpec{
+					OS: &v1.PodOS{
+						Name: "windows", // explicitly set the pod OS to a wrong but valid value
+					},
+					Containers: []v1.Container{
+						{
+							Name:  "wrong-pod-os",
+							Image: imageutils.GetPauseImageName(),
+						},
+					},
+					NodeName: linuxNode.Name, // Set the node to an node which doesn't support
+				},
+			}
+
+			pod = e2epod.NewPodClient(f).Create(ctx, pod)
+			// Check the pod is still not running
+			err = e2epod.WaitForPodFailedReason(ctx, f.ClientSet, pod, "PodOSNotSupported", f.Timeouts.PodStartShort)
+			framework.ExpectNoError(err)
+
+			gotPod, err := f.ClientSet.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+			// Check the pod status is set correctly
+			expectedStatus := pod.Status.DeepCopy()
+			expectedStatus.Phase = gotPod.Status.Phase
```

Comment:
Add a comment before this block saying something like:

```suggestion
                         // overwriting all fields that we expect are overriden by kubelet.
                         // All other fields must stay the same as before kubelet touched them
			expectedStatus.Phase = gotPod.Status.Phase
```